lib/formatters/kakoune.js

```diff
@@ -0,0 +1,39 @@
+import kakouneFormatter from 'eslint-formatter-kakoune';
+
+import Linter from '../linter.js';
+
+export default class KakouneFormatter {
+  constructor(options = {}) {
+    this.options = options;
+    this.console = options.console || console;
+  }
+
+  format(results) {
+    let filteredErrors = {};
+    for (let filePath of Object.keys(results.files)) {
+      let fileErrors = results.files[filePath].messages;
+
+      let errorsFiltered = fileErrors.filter((error) => error.severity === Linter.ERROR_SEVERITY);
+      let warnings = this.options.quiet
+        ? []
+        : fileErrors.filter((error) => error.severity === Linter.WARNING_SEVERITY);
+
+      filteredErrors[filePath] = [...errorsFiltered, ...warnings];
+    }
+
+    results = Object.entries(filteredErrors).map(([filePath, messages]) => {
+      return {
+        filePath,
+        messages: [...messages].map((message) => {
+          return {
+            ...message,
+            // For some reason the column is 0 if the error occurs on the first line and this breaks kakoune
```

Comment:
Will it successfully highlight the violation in the correct location or off-by-one?

Reply:
I`ll test this again.

Reply:
This works as expected

Reply:
Can you update or remove the comment then assuming it works properly?

Reply:
Done